torch/nn/parallel/_functions.py

```diff
@@ -51,12 +53,22 @@ def forward(ctx, target_device, dim, *inputs):
         ctx.target_device = target_device
         ctx.dim = dim
         ctx.input_gpus = tuple(map(lambda i: i.get_device(), inputs))
+        if all(t.dim() == 0 for t in inputs) and dim == 0:
+            inputs = tuple(t.view(1) for t in inputs)
+            warnings.warn('When gathering along dimension 0, all input tensors '
```

Comment:
This seems a little oblique. Maybe something like, "You asked me to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector."